private-templates-service/server/src/db/providers/InMemoryDBProvider.ts

```diff
@@ -0,0 +1,249 @@
+                commitFunction();
+            }
+        } else {
+            logger.error("Object with id: " + doc._id + " already exists. Insertion failed");
+        }
+    }
+
+    private async _writeData(path: string, data: string, successMessage: string, options: Object) : Promise<void> {
+        return fs.writeFile(path, data, options)
+        .then(result => console.log(successMessage))
+        .catch(err => {
+            if(this.writeTryCount < MAX_TRY_COUNT) {
+                logger.error(sprintf("Cannot write to %s. Erros is %s.", path, err));
+                // TODO: set timeout to retry write
+                // setTimeout(this._writeData, 1000); 
+            }
+        });
+    }
+
+    private async _readData(path:string, successMessage: string, options: Object) : Promise<string | void> {
+        return fs.readFile(path, options)
+        .then(buffer => {return buffer.toString()})
+        .catch(err => {
+            logger.error(sprintf("Cannot read from %s. Erros is %s.", path, err));
+        });
+    }
+
+
+    private async _importCollection<T extends Interface.IUser | Interface.ITemplate>
+                                    (path: string, successMessage: string, fromJSON: (j: string) => T | null, collection: Map<string, T>) : Promise<void>{
+        return this._readData(path, successMessage, options)
+        .then(data => {
+            if(data) {
+                let documents: Array<string> = JSON.parse(data);
+                documents.forEach(doc => {
+                    console.log("Inserting: ");
+                    console.log(doc);
+                    let docObject: T | null = fromJSON(JSON.stringify(doc));
+                    if(docObject){
+                        this._insert<T>(docObject, collection, undefined, false);
+                    } else {
+                        throw(new Error("Error when importing a document: cannot parse a document due to wrong JSON structure."))
+                    }
+                });
+            }
+        })
+        .catch(err => logger.error(err));
+    }
+
+    private _checkID<T extends Interface.ITemplate | Interface.IUser>(doc: T){
+        if(!doc._id) {
+            doc._id = generateUniqueID();
+        }
+    }
+
+
+    private async _commitCollection<T extends Interface.IUser | Interface.ITemplate>
+                                    (path: string, successMessage: string, collection: T[], options: Object) : Promise<void>{
+        
+        let data: string = JSON.stringify(Array.from(collection));
+        return this._writeData(path, data, successMessage, options);
+
+    }
+
+    private async _commitUsers() : Promise<void> {
+        let data: Array<Interface.IUser> = Array.from(this.users.values());
+        return this._commitCollection(this.usersPath,
+                        sprintf(SUCCES_WRITE_MESSAGE, USER_COLLECTION_NAME), data, options);
+    }
+
+    private async _commitTemplates() : Promise<void>{
```

Comment:
If you use ES6 arrow function notations, these methods will automatically bind: _commitTempaltes = async (): Promise<void> => {